Source/WebCore/Modules/WebGPU/GPUTexture.h

```diff
@@ -60,6 +63,7 @@ class GPUTexture : public RefCounted<GPUTexture> {
     }
 
     Ref<PAL::WebGPU::Texture> m_backing;
+    mutable Vector<std::pair<std::optional<GPUTextureViewDescriptor>, Ref<GPUTextureView>>> m_cachedTextureViews;
```

Comment:
I think this is conceptually backwards from an ownership perspective. Objects can hold strong references to their parent object, not the other way around.

E.g. retaining a device shouldn't retain all objects that have ever been created from that device; instead, if a JS application has created all the objects they'll need, they should be free to drop all references to the Device, but the device can stay alive anyway because it can have child objects that retain it. So, the ownership model is inverted, and branch objects shouldn't retain the leaf objects they've created; instead, the leafs can retain the branches.